inference-engine/tests/functional/plugin/shared/src/subgraph_tests/input_conv.cpp

```diff
@@ -0,0 +1,110 @@
+    result << "targetDevice=" << targetDevice;
+    for (auto const& configItem : configuration) {
+        result << "_configItem=" << configItem.first << "_" << configItem.second;
+    }
+    return result.str();
+}
+
+InferenceEngine::Blob::Ptr InputConvTest::GenerateInput(const InferenceEngine::InputInfo& info) const {
+    InferenceEngine::Blob::Ptr blob = make_blob_with_precision(info.getTensorDesc());
+    blob->allocate();
+    auto precision = info.getPrecision();
+
+    auto* rawBlobDataPtr = blob->buffer().as<float*>();
+    auto counter = 0;
+    for (size_t i = 0; i < blob->size(); i++) {
+        auto value = counter;
+        if (typeid(precision) == typeid(typename InferenceEngine::PrecisionTrait<InferenceEngine::Precision::FP16>::value_type)) {
+            rawBlobDataPtr[i] = ngraph::float16(value).to_bits();
+        } else {
+            rawBlobDataPtr[i] = value;
+        }
+
+        if (counter == 15)
+            counter = 0;
+        else
+            ++counter;
+    }
+    return blob;
+}
+
+void InputConvTest::SetUp() {
+    auto generateWeights = [](std::size_t out_channels, std::size_t kernel_size) {
+        std::vector<float> res;
+        for (int i = 0; i < out_channels; ++i) {
+            for (int j = 0; j < kernel_size; ++j) {
+                res.emplace_back(j == 0 ? 1 : 0);
+            }
+        }
+
+        return res;
+    };
+
+    InferenceEngine::Precision netPrecision;
+    std::map<std::string, std::string> tempConfig;
+    std::vector<size_t> inputShape;
+    std::tie(netPrecision, targetDevice, tempConfig, inputShape) = this->GetParam();
+    configuration.insert(tempConfig.begin(), tempConfig.end());
+
+    auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+    size_t output_channels = 4;
+    size_t kernel_y = 1;
+    size_t kernel_x = 9;
+    auto params = ngraph::builder::makeParams(ngPrc, { inputShape });
+
+    auto conv_0 = ngraph::builder::makeConvolution(params[0], ngPrc, { kernel_y, kernel_x }, { 1, 1 }, { 0, 0 },
+        { 0, 0 }, { 1, 1 }, ngraph::op::PadType::VALID, output_channels, false, generateWeights(output_channels, kernel_x));
+
+    //// without additional op at the end for int16 ConvertToFloat gives incorrect values but intermediate representation is ok
+    //// fp32 works with and without reshape
+    //size_t num_output_width = (((inputShape[1] * inputShape[2] * inputShape[3] - kernel_x * kernel_y) / inputShape[1]) + 1);
+    //std::vector<size_t> outFormShapes_0 = { 1, output_channels * num_output_width };
+    //auto pattern_0 = std::make_shared<ngraph::opset1::Constant>(ngraph::element::Type_t::i64, ngraph::Shape{ 2 }, outFormShapes_0);
+    //auto reshape_0 = std::make_shared<ngraph::opset1::Reshape>(mul_0, pattern_0, false);
+
+    auto permute_0 = std::make_shared<ngraph::op::Transpose>(conv_0,
+        ngraph::op::Constant::create(ngraph::element::i64, ngraph::Shape{ 4 }, { 0, 3, 1, 2 }));
+    permute_0->set_friendly_name("permute1");
```

Comment:
call it permute0 if you want for some reason set the name at all so the variable name correcponds to its friendly name